### PR TITLE
chore: improve the validations of messages

### PR DESCRIPTION
**Pull request by forcodedancing**
### Description

Improve the basic validation of messages.

### Rationale

Chore

### Example

NA

### Changes

Notable changes: 
* message validations

### CODE REVIEW

File: x/permission/types/types.go [hunk 1]
```diff
@@ -167,26 +183,25 @@ func (s *Statement) ValidateBasic(resType resource.ResourceType) error {
...
-		containsCreateObject := false
-		for _, a := range s.Actions {
-			if !BucketAllowedActions[a] {
-				return ErrInvalidStatement.Wrapf("%s not allowed to be used on bucket.", a.String())
-			}
-			if a == ACTION_CREATE_OBJECT {
-				containsCreateObject = true
-			}
-		}
+		//containsCreateObject := false
```
Comment (Fynnss) [inline x/permission/types/types.go:186]: You'd better to remove these useless code, rather than comment.

File: x/permission/types/types.go [hunk 2]
```diff
@@ -167,26 +183,25 @@ func (s *Statement) ValidateBasic(resType resource.ResourceType) error {
...
+		//		return ErrInvalidStatement.Wrapf("%s not allowed to be used on bucket.", a.String())
+		//	}
+		//	if a == ACTION_CREATE_OBJECT {
+		//		containsCreateObject = true
+		//	}
+		//}
+		//if !containsCreateObject && s.LimitSize != nil {
+		//	return ErrInvalidStatement.Wrap("The LimitSize option can only be used with CreateObject actions at the bucket level. .")
+		//}
 		for _, r := range s.Resources {
```
Comment (Fynnss) [inline x/permission/types/types.go:198]: Shall we move the GRN check to `ValidateRuntime` ? I think yes.

File: x/permission/types/types.go (hunk 1 above)
Comment (forcodedancing) [reply x/permission/types/types.go:186]: Sure.

File: x/permission/types/types.go (hunk 2 above)
Comment (forcodedancing) [reply x/permission/types/types.go:198]: Why we need to move the GRN check to ValidateRuntime? 
It does not between these upgrades. Keep it in `ValidateBasic` will be useful for CLI.

Comment (Fynnss) [reply x/permission/types/types.go:186]: Because it may be adjusted in the future.

File: x/permission/types/types.go [hunk 3]
```diff
@@ -167,26 +183,13 @@ func (s *Statement) ValidateBasic(resType resource.ResourceType) error {
 	case resource.RESOURCE_TYPE_UNSPECIFIED:
 		return ErrInvalidStatement.Wrap("Please specify the ResourceType explicitly. Not allowed set RESOURCE_TYPE_UNSPECIFIED")
 	case resource.RESOURCE_TYPE_BUCKET:
-		containsCreateObject := false
```
Comment (j75689) [inline x/permission/types/types.go:170]: maybe moving all `ValidaeBasic` function to `ValidateRuntime` would be better? 
It's kind of complex and unreadable if you split it into two place..

Comment (forcodedancing) [reply x/permission/types/types.go:170]: `ValidaeBasic` is still used by cli. If we remove it, there will be no validation for cli.

File: x/permission/types/types.go [hunk 4]
```diff
@@ -211,30 +214,48 @@ func (s *Statement) ValidateBasic(resType resource.ResourceType) error {
...
+		case resource.RESOURCE_TYPE_OBJECT:
+			if s.Resources != nil {
+				return ErrInvalidStatement.Wrap("The Resources option can only be used at the bucket level. ")
+			}
+		case resource.RESOURCE_TYPE_GROUP:
+			if s.Resources != nil {
+				return ErrInvalidStatement.Wrap("The Resources option can only be used at the bucket level. ")
 			}
+		default:
+			return ErrInvalidStatement.Wrap("unknown resource type.")
```
Comment (j75689) [inline x/permission/types/types.go:236]: that will always return here if `Nagqu` upgrade was applied.

File: x/permission/types/types.go [hunk 5]
```diff
@@ -211,30 +214,48 @@ func (s *Statement) ValidateBasic(resType resource.ResourceType) error {
...
-	switch resType {
-	case resource.RESOURCE_TYPE_UNSPECIFIED:
-		return ErrInvalidStatement.Wrap("Please specify the ResourceType explicitly. Not allowed set RESOURCE_TYPE_UNSPECIFIED")
-	case resource.RESOURCE_TYPE_BUCKET:
-		for _, r := range s.Resources {
-			_, err := regexp.Compile(r)
-			if err != nil {
-				return ErrInvalidStatement.Wrapf("The Resources regexp compile failed, err: %s", err)
+func (s *Statement) ValidateRuntime(ctx sdk.Context, resType resource.ResourceType) error {
+	if ctx.IsUpgraded(upgradetypes.Nagqu) {
```
Comment (zjubfd) [inline x/permission/types/types.go:218]: we delete `if s.Effect == EFFECT_UNSPECIFIED`, will it cause consensus issue?

Comment (forcodedancing) [reply x/permission/types/types.go:218]: `validateBasic` has covered this, i think.

**Review by zjubfd — APPROVED** [review]

**Pull request merged**